src/Microsoft.OData.Edm/Schema/EdmModelBase.cs

```diff
@@ -173,25 +199,39 @@ public virtual IEnumerable<IEdmOperation> FindDeclaredBoundOperations(IEdmType b
         /// </returns>
         public virtual IEnumerable<IEdmOperation> FindDeclaredBoundOperations(string qualifiedName, IEdmType bindingType)
         {
-            IEnumerable<IEdmOperation> enumerable = this.FindDeclaredOperations(qualifiedName);
-            IList<IEdmOperation> operations = enumerable as IList<IEdmOperation>;
-            if (operations != null)
+            IEnumerable<IEdmOperation> enumarable = this.FindDeclaredBoundOperations(bindingType);  
+            
+            if(enumarable == null || !enumarable.Any())

```

Comment:
>|| !enumarable.Any()) [](start = 34, length = 21)

no; don't use enumerable.Any() -- that's linq, and is expensive.  Instead, optimize for common case by checking to see if enumerable is an IList and, if so, using it as such.